Extension/lib/content-script/websocket.js

```diff
@@ -14,337 +14,419 @@
+         *
+         * @param wrapper       Wrapper instance
+         * @param prop          Property name
+         * @param defaultValue  Default value to be returned if real WS does not yet exist
+         * @returns {*}
+         */
+        var getWrappedProperty = function (wrapper, prop, defaultValue) {
+            var wrapped = map.get(wrapper);
+            if (!wrapped) {
+                return defaultValue;
+            }
 
-        if (wrapped instanceof Wrapped) {
-            wrapped[prop] = value;
-        } else {
-            wrapped.properties[prop] = value;
-        }
-    };
+            if (wrapped instanceof Wrapped) {
+                return wrapped[prop];
+            }
 
-    /**
-     * Fake websocket constructor
-     *
-     * @param url
-     * @param protocols
-     * @constructor
-     */
-    var WebSocket = function (url, protocols) {
-        'native';
-        if (window.location.protocol === 'https:'
-            && url.lastIndexOf('ws:', 0) === 0) {
-            var ws = new Wrapped(url, protocols);
-            if (ws) {
-                ws.close();
+            return wrapped.properties.hasOwnProperty(prop) ?
+                wrapped.properties[prop] :
+                defaultValue;
+        };
+
+        /**
+         * Sets a value of the specified property.
+         *
+         * @param wrapper   Wrapper instance
+         * @param prop      Property name
+         * @param value     Value to set
+         */
+        var setWrappedProperty = function (wrapper, prop, value) {
+            if (value instanceof Function) {
+                value = value.bind(wrapper);
             }
-        }
 
-        Object.defineProperties(this, {
-            'binaryType': {
-                get: function () {
-                    return getWrappedProperty(this, 'binaryType', 'blob');
-                },
-                set: function (value) {
-                    setWrappedProperty(this, 'binaryType', value);
+            var wrapped = map.get(wrapper);
+            if (!wrapped) {
+                return;
+            }
+
+            if (wrapped instanceof Wrapped) {
+                wrapped[prop] = value;
+            } else {
+                wrapped.properties[prop] = value;
+            }
+        };
+
+        /**
+         * Fake websocket constructor.
+         * https://developer.mozilla.org/en-US/docs/Web/API/WebSocket
+         * 
+         * Original websocket opens a connection to a specified url. This wrapper works in a 
+         * different way, it first checks if this WS connection should be blocked or not.
+         *
+         * @param url       The URL to which to connect
+         * @param protocols Either a single protocol string or an array of protocol strings
+         * @constructor
+         */
+        var WebSocket = function (url, protocols) {
+            'native';
+            if (window.location.protocol === 'https:' &&
+                    url.lastIndexOf('ws:', 0) === 0) {
```

Comment:
wss is allowed on https pages.

The only purpose of this check is to let original WebSocket throw a "native" error.